src/librustc_interface/passes.rs

```diff
@@ -667,13 +669,28 @@ fn write_out_deps(sess: &Session, outputs: &OutputFilenames, out_filenames: &[Pa
     let result = (|| -> io::Result<()> {
         // Build a list of files used to compile the output and
         // write Makefile-compatible dependency rules
-        let files: Vec<String> = sess.source_map()
+        let mut files: Vec<String> = sess.source_map()
             .files()
             .iter()
             .filter(|fmap| fmap.is_real_file())
             .filter(|fmap| !fmap.is_imported())
             .map(|fmap| escape_dep_filename(&fmap.name))
             .collect();
+
+        for cnum in compiler.cstore.crates_untracked() {
```

Comment:
Does this include all crates in library search paths, or only the ones brought in via `extern crate`?

Reply:
I believe it'll be only those "used" (extern crate, use, etc)

Reply:
Can you add a test confirming this?

Reply:
I think such a test would be relatively complicated (probably some sort of run-make, I guess) but basically the `crates_untracked()` bit here starting to pick up all tests would be noticed by plenty of other tests AFAIK since it interacts with incremental, etc. too. I have confirmed manually that not all crates in the sysroot and other search paths show up.